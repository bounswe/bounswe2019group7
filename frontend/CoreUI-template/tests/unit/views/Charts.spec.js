import Vue from 'vue'
import { shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue'
import Charts from '@/views/Charts'

Vue.use(BootstrapVue)

describe('Charts.vue', () => {
  it('has a name', () => {
    expect(Charts.name).toMatch('charts')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Charts)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Charts', () => {
    const wrapper = shallowMount(Charts)
    expect(wrapper.is(Charts)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Charts)
    expect(wrapper.element).toMatchSnapshot()
  })
})
