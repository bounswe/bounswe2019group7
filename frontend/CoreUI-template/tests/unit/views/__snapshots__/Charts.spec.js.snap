// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts.vue renders correctly 1`] = `
<div
  class="animated fadeIn"
>
  <b-card-group-stub
    class="card-columns"
    columns="true"
    tag="div"
  >
    <b-card-stub
      bodytag="div"
      footertag="div"
      header="Line Chart"
      headertag="div"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        class="chart-wrapper"
      >
        <line-example-stub
          chartid="chart-line-01"
          cssclasses=""
          height="400"
          plugins=""
          width="400"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="div"
      header="Bar Chart"
      headertag="div"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        class="chart-wrapper"
      >
        <bar-example-stub
          chartid="chart-bar-01"
          cssclasses=""
          height="400"
          plugins=""
          width="400"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="div"
      header="Doughnut Chart"
      headertag="div"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        class="chart-wrapper"
      >
        <doughnut-example-stub
          chartid="chart-doughnut-01"
          cssclasses=""
          height="400"
          plugins=""
          width="400"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="div"
      header="Radar Chart"
      headertag="div"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        class="chart-wrapper"
      >
        <radar-example-stub
          chartid="chart-radar-01"
          cssclasses=""
          height="400"
          plugins=""
          width="400"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="div"
      header="Pie Chart"
      headertag="div"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        class="chart-wrapper"
      >
        <pie-example-stub
          chartid="chart-pie-01"
          cssclasses=""
          height="400"
          plugins=""
          width="400"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="div"
      header="Polar Area Chart"
      headertag="div"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        class="chart-wrapper"
      >
        <polar-area-example-stub
          chartid="chart-polar-area-01"
          cssclasses=""
          height="400"
          plugins=""
          width="400"
        />
      </div>
    </b-card-stub>
  </b-card-group-stub>
</div>
`;
